source/signaling_api.c

```diff
@@ -434,6 +434,171 @@ SignalingResult_t Signaling_ParseDescribeSignalingChannelResponse( const char *
 
 /*-----------------------------------------------------------*/
 
+SignalingResult_t Signaling_ConstructSessionTokenCredentialsRequest( char * endpoint,
+                                                                     char * roleAlias,
+                                                                     SignalingRequest_t * pRequestBuffer )
+{
+    SignalingResult_t result = SIGNALING_RESULT_OK;
+    int snprintfRetVal = 0;
+
+    if( ( endpoint == NULL ) ||
+        ( roleAlias == NULL ) ||
+        ( pRequestBuffer == NULL ) ||
+        ( pRequestBuffer->pUrl == NULL ) )
+    {
+        result = SIGNALING_RESULT_BAD_PARAM;
+    }
+
+    if( result == SIGNALING_RESULT_OK )
+    {
+        snprintfRetVal = snprintf( pRequestBuffer->pUrl,
+                                   pRequestBuffer->urlLength,
+                                   "https://%s/role-aliases/%s/credentials",
+                                   endpoint,
+                                   roleAlias );

```

Comment:
```suggestion
                                   "https://%.*s/role-aliases/%.*s/credentials",
                                   ( int ) endpointLength,
                                   endpoint,
                                   ( int ) roleAliasLength,
                                   roleAlias );
```
Please follow the pattern on other APIs to ask for the length of endpoint and role alias?
- example: https://github.com/awslabs/amazon-kinesis-video-streams-signaling/blob/d828c58802b77e2e60feb242965350bb729f31bc/source/signaling_api.c#L228-L232

Reply:
It came to my mind also, when I learned what is advantage of %s and %.*s , it showed that the later one is used when string contains much more data then we need, so we specify the length. Till where you need the length. So shall I do it for the similarity of other API's or does it have a advantage in including the length ?

Reply:
Using only `char*` is restrictive and error-prone. A better approach is to use a pointer-length pair.